test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -2156,26 +2190,52 @@ static int traffic_mngr_suite_init(void)
 	if (ret > 0)
 		goto skip_tests;
 
+	odp_tm_egress_init(&egress);
+	egress.egress_kind = ODP_TM_EGRESS_PKT_IO;
+	egress.pktio = xmt_pktio;
+
+	/* Get TM capabilities */
+	ret = odp_tm_egress_capabilities(&egress_capa, &egress);
+	if (ret) {
+		ODPH_ERR("Failed to retrieve tm capabilities");
+		return ret;
+	}
+
+	/* Check for sufficient TM queues */
+	if (egress_capa.max_tm_queues < NUM_TM_QUEUES)
```

Comment:
This is inconsistent with create_tm_system() that sets requirements.max_tm_queues to NUM_TM_QUEUES + 1. I do not know why it does it and I think it is bad that NUM_TM_QUEUES does not really mean the total number of TM queues (or if it does then the +1 is there for no reason), so maybe the best fix is not to change this to NUM_TM_QUEUES + 1 but something else.

Reply:
Reduced it to NUM_TM_QUEUES at requirements init.